modules/renter/upload.go

```diff
@@ -102,6 +103,10 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 	lockID := r.mu.Lock()
 	defer r.mu.Unlock(lockID)
 
+	if filepath.Ext(up.Filename) != filepath.Ext(up.Nickname) {
+		return errors.New("nickname and file name must have the same extension")
+	}
```

Comment:
Yeah, I guess that would be an unfortunate scenario, even though I think it's unlikely. My preference would be for the UI to warn you about that possibility if you change the extension. I just don't want to restrict people who know what they're doing.